### DISCUSSION TITLE
how to make opertions in for loop parallel?

### DISCUSSION THREAD

**Discussion by whsasf** [fastapi/fastapi#9926] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def xx:
     for i in range(len(paper_fragments)):
        print(f"{i}")
        embedd = await get_embedding(paper_fragments[i])
        print(f"embedding {i} begin...")
        x = time.time()
        abstract = json.loads(await get_abstract(paper_fragments[i], 200))
        print(f"embedding {i} finished ...")
        y = time.time()
        print(f"time span {i}: {y-x}")
```

### Description

like the code, i have a for loop in a async def, currently it gives me the output like this :
-----
 0
embedding 0 begin...
embedding 0 finished ...

 1
embedding 1 begin...
embedding 1 finished ...
--------

but i want it run parallel and give me output like this 
----
0
1
embedding 0 begin...
embedding 1 begin...
embedding 0 finished ...
embedding 1 finished ...
----

i have tried to create a new loop, but it doesn't work wuth uvloop.

Do you have other way to achieve this ? thanks.

### Operating System

Linux, macOS

### FastAPI Version

0.98.0

### Python Version

Python 3.9.6

**Comment by n8sty** [comment, marked as answer]
This isn't about FastAPI but rather async programming in Python. Look at the docs for `asyncio.gather` (https://docs.python.org/3/library/asyncio-task.html#asyncio.gather) and `asyncio.TaskGroup.create_task` (https://docs.python.org/3/library/asyncio-task.html#asyncio.TaskGroup.create_task) for a couple of ways of running tasks concurrently. There are other interfaces for parallel and concurrent programming in Python, but I'm assuming you want something asynchronous.

**Comment by whsasf** [comment]
the answer really help. thank you very much.
